tests/Signals_perftest.m

```diff
@@ -1,90 +1,191 @@
 classdef Signals_perftest < matlab.perftest.TestCase
+% Runs performance tests for Signals
+%
+% Performs benchmarking for various operations on signals: 
+% nops, assignment ops, logical ops, scalar arithmetic ops, listener ops,
+% and signal methods: `map`, `scan`, `identity`, `subscriptable`, etc...
+% (via `test_mainOps` and `test_signalsOps` methods)
+%
+% And performs benchmarking for propogations through each signal in 
+% networks of various widths and depths (30 to 1000 nodes spread across 2
+% to 20 layers) (via `test_networkOps` method) (*Note: for reference, the 
+% `docs\examples\advancedChoiceWorld` exp def has 338 nodes over 10 layers)
+%
+% In this class, the 'Net' property is a Signals network, in which the 
+% 'A' & 'B' properties are treated as "input layer" signals from which all 
+% other signals are defined.
+%   
+% Example - display the mean measured time for running each test:
+%   results = runperf('Signals_perftest2.m');
+%   fullTableResults = vertcat(results.Samples);
+%   meanTimeByTestTable = varfun(@mean, fullTableResults,...
+%     'InputVariables', 'MeasuredTime', 'GroupingVariables', 'Name')
   
   properties
-    net
-    A
-    B
-    C
-    func
-    verify = false
-    results
-    reps = 2000
+    Net % Signals network ('sig.Net' object)
+    A % origin signal ('sig.node.OriginSignal' object)
+    B % origin signal ('sig.node.OriginSignal' object)
+    Reps = 1000 % number of test repetitions
+    MexOps % operations which are run directly in compiled mexnet
   end
   
   properties (TestParameter)
-    ops = num2cell([1:6 10:14 30 40 50])
+    % main operations to test (whose results are posted in the paper)
+    MainOps = {@post, @gt, @ge, @lt, @le, @eq, @plus, @minus, @times, ... 
+      @rdivide, @map, @scan, @subscriptable, @onValue}
+    % logical/arithmetical subset of operations to test
+    BasicOps = {@gt, @ge, @lt, @le, @eq, @plus, @minus, @times, @rdivide}
+    %those operations which are run directly in compiled mexnet
+    mexOpKeys = num2cell([1:6 10:14 30 40])
```

Comment:
mexOpKeys is unused and doesn't relate to any of the other properties, I think you missed the point of the original containers map...

Reply:
hm? it's used in `createNetwork`?

Reply:
a new function can be added to only test `testCase.MexOps` if that's what you want?

Reply:
> hm? it's used in `createNetwork`?

Ah yes so it is, I did a quick search but must have made a typo or something

Reply:
The question is what do we want to separate out.  I'd argue that because the MEX operations should have a different time range and be run by very different code, it's worth having the option of testing them separately.  Then while making changes to the C code we can compile and run only the relevant tests.  

Personally I think we should go with Chris's suggestion and use and experiment definition to bench mark.  Why pick a handful of random methods (the ones you call 'main') when we can test all of them, along with a so-called real world example experiment function, then compare all the results with previous commits, etc.?

Reply:
Ok, so the properties to keep are `MexOps` and `SignalsOps`, and the subset of functions tested in the paper will just be pulled from these two within a test function